tests/pamiq_core/torch/test_model.py

```diff
@@ -0,0 +1,125 @@
+import copy
+from threading import RLock
+from typing import Any, Protocol, override
+
+import pytest
+import torch
+import torch.nn as nn
+
+from pamiq_core.model import InferenceModel, TrainingModel
+from pamiq_core.torch import (
+    TorchInferenceModel,
+    TorchTrainingModel,
+    default_infer_procedure,
+)
+
+
+class TestTorchInferenceModel:
+    @pytest.fixture
+    def model(self) -> nn.Module:
+        return nn.Linear(3, 5)
+
+    @pytest.fixture
+    def torch_inference_model(self, model: nn.Module) -> TorchInferenceModel:
+        torch_inference_model = TorchInferenceModel(model, default_infer_procedure)
+        return torch_inference_model
+
+    def test_raw_model(
+        self, torch_inference_model: TorchInferenceModel, model: nn.Module
+    ) -> None:
+        assert model is torch_inference_model.raw_model
+
+    def test_infer(
+        self, torch_inference_model: TorchInferenceModel, model: nn.Module
+    ) -> None:
+        input_tensor = torch.randn([2, 3])
+        output_tensor = torch_inference_model.infer(input_tensor)
+        expected_tensor = model(input_tensor)
+        assert torch.equal(output_tensor, expected_tensor)
+
+
+@pytest.mark.parametrize(
+    ["has_inference_model", "inference_thread_only"],
+    [[True, True], [True, False], [False, False]],
+)
+class TestTorchTrainingModel:
+    @pytest.fixture
+    def model(self) -> nn.Module:
+        return nn.Linear(3, 5)
+
+    @pytest.fixture
+    def torch_training_model(
+        self, model: nn.Module, has_inference_model: bool, inference_thread_only: bool
+    ) -> TorchTrainingModel:
+        torch_training_model = TorchTrainingModel(
+            model, has_inference_model, inference_thread_only
+        )
+        return torch_training_model
```

Comment:
inference_thread_only用のfixtureを定義した方が良い。
```suggestion
    @pytest.fixture
    def torch_training_model(
        self, model: nn.Module
    ) -> TorchTrainingModel:
        return TorchTrainingModel(
            model, True, False
        )

    @pytest.fixture
    def torch_training_model_inference_only(
        self, model: nn.Module) -> TorchTrainingModel:
        return TorchTrainingModel(
            model, True, True
        )
```